packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/fragment.js

```diff
@@ -9,58 +10,85 @@ import { build_template_literal, build_update } from './utils.js';
  * @param {boolean} is_element
  * @param {ComponentContext} context
  */
-export function process_children(nodes, expression, is_element, { visit, state }) {
+export function process_children(nodes, initial, is_element, { visit, state }) {
 	const within_bound_contenteditable = state.metadata.bound_contenteditable;
+	let prev = initial;
+	let skipped = 0;
 
 	/** @typedef {Array<Text | ExpressionTag>} Sequence */
-
 	/** @type {Sequence} */
 	let sequence = [];
 
+	/** @param {boolean} check */
+	function get_node(check) {
+		if (skipped === 0) {
+			return prev(check);
+		}
+
+		return b.call(
+			'$.sibling',
+			prev(false),
+			(check || skipped !== 1) && b.literal(skipped),
+			check && b.true
+		);
+	}
+
+	/**
+	 * @param {boolean} check
+	 * @param {string} name
+	 */
+	function flush_node(check, name) {
+		const expression = get_node(check);
+		let id = expression;
+
+		if (id.type !== 'Identifier') {
+			id = b.id(state.scope.generate(name));
+			state.init.push(b.var(id, expression));
+		}
+
+		prev = () => id;
+		skipped = 1;
+
+		return id;
+	}
+
 	/**
 	 * @param {Sequence} sequence
 	 */
 	function flush_sequence(sequence) {
-		if (sequence.length === 1) {
-			const node = sequence[0];
-
-			if (node.type === 'Text') {
-				let prev = expression;
-				expression = () => b.call('$.sibling', prev(false));
-				state.template.push(node.raw);
-				return;
-			}
+		if (sequence.length === 1 && sequence[0].type === 'Text') {
+			skipped += 1;
+			state.template.push(sequence[0].raw);
+			return;
 		}
 
-		// if this is a standalone `{expression}`, make sure we handle the case where
-		// no text node was created because the expression was empty during SSR
-		const needs_hydration_check = sequence.length === 1;
-		const id = get_node_id(expression(needs_hydration_check), state, 'text');
-
 		state.template.push(' ');
 
 		const { has_state, has_call, value } = build_template_literal(sequence, visit, state);
 
-		const update = b.stmt(b.call('$.set_text', id, value));
+		// if this is a standalone `{expression}`, make sure we handle the case where
+		// no text node was created because the expression was empty during SSR
+		const needs_hydration_check = sequence.length === 1;
+		const id = flush_node(needs_hydration_check, 'text');
 
-		if (has_call && !within_bound_contenteditable) {
-			state.init.push(build_update(update));
-		} else if (has_state && !within_bound_contenteditable) {
-			state.update.push(update);
-		} else {
+		if (within_bound_contenteditable || !has_state) {
 			state.init.push(b.stmt(b.assignment('=', b.member(id, 'nodeValue'), value)));
-		}
+		} else {
+			const update = b.stmt(b.call('$.set_text', id, value));
 
-		expression = (is_text) => b.call('$.sibling', id, is_text && b.true);
+			if (has_call && !within_bound_contenteditable) {
+				state.init.push(build_update(update));
+			} else if (has_state && !within_bound_contenteditable) {
+				state.update.push(update);
+			}
+		}
```

Comment:
This can be simplified to

```suggestion
		if (within_bound_contenteditable || !has_state) {
			state.init.push(b.stmt(b.assignment('=', b.member(id, 'nodeValue'), value)));
		} else if (!within_bound_contenteditable) {
			const update = b.stmt(b.call('$.set_text', id, value));

			if (has_call) {
				state.init.push(build_update(update));
			} else if (has_state) {
				state.update.push(update);
			}
        }
```

... which shows that there's now the possibility to not add anything to init or update at all - previously it always did. Is that a bug?

Reply:
huh, whoops — just wanted to avoid creating `update` unnecessarily. reverted this drive-by change.

i notice that if we remove the contenteditable stuff altogether no tests fail, which seems weird. making a mental note to look into this

Reply:
Maybe @trueadm knows more, IIRC he worked on content editable a while back

Reply:
None of the contentEditable stuff is very testable unfortunately. JSDOM has no support for it and Playwright support is flaky (from experience using it with Lexical).

Reply:
ah, i see. we should probably add a snapshot test in that case. will make a note